frontend/ui/elements/screensaver_menu.lua

```diff
@@ -153,12 +154,23 @@ return {
                         end,
                     },
                     {
-                        text = _("Stretch covers and images to fit screen"),
+                        text_func = function()
+                            if G_reader_settings:nilOrFalse("screensaver_stretch_images") then
+                                return _("Stretch to fit screen")
+                            elseif G_reader_settings:readSetting("screensaver_stretch_limit_percentage") then
+                                return _("Stretch to fit screen (with limit)")
+                            else
+                                return _("Stretch to fit screen")
+                            end
+                        end,
                         checked_func = function()
                             return G_reader_settings:isTrue("screensaver_stretch_images")
                         end,
-                        callback = function()
-                            G_reader_settings:toggle("screensaver_stretch_images")
+                        help_text_func = function()
+                            return _("If the image and the screen have similar aspect ratios, stretch the image instead of keeping its aspect ratio.")
```

Comment:
The first part of the sentence is only valid with your option.
If the spin dialog is clear enough, may be it doesn't need any help_text ?